### PR TITLE
fix(openclaw_gateway): scope generated session keys by agent

### LINKED ISSUE #1205: openclaw_gateway adapter: session keys missing agent:{id}: prefix for multi-agent routing
## Problem

When using the `openclaw_gateway` adapter with multiple agents, the adapter generates session keys like `paperclip:issue:{issueId}` or `paperclip:run:{runId}`. However, the OpenClaw gateway requires session keys prefixed with `agent:{agentId}:` for multi-agent routing (e.g. `agent:cody:paperclip:issue:{id}`).

When the key lacks this prefix, the gateway defaults to the "main" agent, causing a mismatch error when `agentParams.agentId` specifies a different agent:

```
invalid agent params: agent "cody" does not match session key agent "main"
```

## Root Cause

`resolveSessionKey()` in `packages/adapters/openclaw-gateway/src/server/execute.ts` (line 129) generates keys without considering the configured `agentId`. The `agentId` config value IS used later (line 1078-1080) to set `agentParams.agentId`, but is never passed to the session key builder.

## Current Workaround

Using `sessionKeyStrategy: "fixed"` with a manually prefixed `sessionKey: "agent:cody:paperclip"` works, but loses per-issue context isolation (all heartbeats for an agent share one session).

## Proposed Fix

1. Add optional `agentId` param to `resolveSessionKey()` input type
2. When `agentId` is provided, prefix the generated key: `agent:${agentId}:${baseKey}`
3. Pass `configuredAgentId` into the resolver call at line 1059

This is backward-compatible — only prefixes when `agentId` is configured. No behaviour change for single-agent setups.

## Environment

- Paperclip: latest main branch (local_trusted mode)
- OpenClaw Gateway: v0.x (token auth, multi-agent)
- 3 agents configured with openclaw_gateway adapter (different agentIds)

### PR DESCRIPTION
**Pull request by larrylhollan**
## Thinking Path

> - Paperclip orchestrates AI agents for zero-human companies.
> - One way Paperclip does that is by invoking external runtimes through adapters, including the `openclaw_gateway` adapter.
> - In multi-agent OpenClaw setups, routing is determined by both `agentId` and agent-scoped session keys.
> - The adapter already forwards `agentId`, but generated `issue` / `run` session keys were still emitted as legacy unscoped keys like `paperclip:issue:<id>`.
> - OpenClaw correctly treats those unscoped keys as belonging to `main`, which causes a mismatch error when the payload also targets a non-main agent such as `eng` or `coo`.
> - This pull request scopes generated `issue` / `run` session keys by agent when a non-main agent is targeted, while preserving explicitly scoped fixed keys as-is.
> - The benefit is that multi-agent OpenClaw routing keeps true per-issue / per-run continuity without forcing users onto the `fixed` workaround that can blur issue context.

## What Changed

- updated `resolveSessionKey()` in the `openclaw_gateway` adapter to accept `agentId`
- generate agent-scoped `issue` / `run` session keys for non-main agents (for example `agent:eng:paperclip:issue:<issueId>`)
- preserve already-scoped fixed session keys so existing workarounds do not get double-prefixed
- keep `main` / legacy behavior unchanged
- added focused regression tests for `eng`, `coo`, `main`, unscoped fixed keys, and already-scoped fixed keys
- clarified the session-routing behavior in the adapter README, including the continuity impact for unscoped `fixed` keys

Fixes #1205.

## Verification

- `pnpm --filter @paperclipai/adapter-openclaw-gateway typecheck`
- `pnpm --filter @paperclipai/adapter-openclaw-gateway build`
- `./node_modules/.bin/vitest run --config .vitest-openclaw-gateway-temp.config.ts`
  - targeted `packages/adapters/openclaw-gateway/src/server/execute.test.ts`
  - **11 tests passing locally**
- validated the same fix on a live local Paperclip instance (`/Users/larry/projects/paperclip`) wired to a live OpenClaw gateway:
  - Engineering issue wake accepted with `agent:eng:paperclip:issue:<issueId>`
  - COO issue wake accepted with `agent:coo:paperclip:issue:<issueId>`
  - the prior `invalid agent params: agent "eng/coo" does not match session key agent "main"` repro no longer occurred

## Risks

- Low risk: behavior changes only when a non-main agent is targeted and the generated session key is not already agent-scoped.
- Existing explicit agent-scoped `fixed` workarounds are preserved unchanged to avoid double-prefixing.
- Main / legacy single-agent behavior remains unchanged.
- Deploy note: unscoped `fixed` keys for non-main agents will begin resolving to agent-scoped keys after this change, so operators may want to drain any in-flight sessions first.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [ ] If this change affects the UI, I have included before/after screenshots
- [x] I have updated relevant documentation to reflect my changes
- [x] I have considered and documented any risks above
- [x] I have addressed current Greptile/reviewer comments and re-ran targeted validation

### CODE REVIEW

File: packages/adapters/openclaw-gateway/src/server/execute.ts [hunk 1]
```diff
@@ -126,16 +126,29 @@ function normalizeSessionKeyStrategy(value: unknown): SessionKeyStrategy {
   return "issue";
 }
 
-function resolveSessionKey(input: {
+function hasAgentScope(sessionKey: string): boolean {
+  return /^agent:[a-z0-9_-]+:.+/i.test(sessionKey.trim());
```
[the remark this answers is not in the document]
Comment (larrylhollan) [reply packages/adapters/openclaw-gateway/src/server/execute.ts]: Implemented — dropped the `/i` flag in `hasAgentScope()` so the regex matches the lowercase agent/session-key shape we actually support.

File: packages/adapters/openclaw-gateway/src/server/execute.test.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  it("returns legacy run key when agentId is null", () => {
+    const key = resolveSessionKey({
+      strategy: "run",
+      configuredSessionKey: null,
+      runId,
+      issueId: null,
+      agentId: null,
+    });
+    expect(key).toBe("paperclip:run:run-456");
```
[the remark this answers is not in the document]
Comment (larrylhollan) [reply packages/adapters/openclaw-gateway/src/server/execute.test.ts:71]: Implemented — added the missing symmetry test for `run` + `agentId: "main"` so it is covered explicitly alongside the existing `null` case.

File: packages/adapters/openclaw-gateway/README.md [hunk 3]
```diff
@@ -43,6 +43,8 @@ The adapter supports the same session routing model as HTTP OpenClaw mode:
 
 Resolved session key is sent as `agent.sessionKey`.
 
+When a non-main `agentId` is present, generated `issue` / `run` session keys are automatically agent-scoped (for example `agent:eng:paperclip:issue:<issueId>`) so multi-agent OpenClaw routing preserves issue/run continuity per agent. Existing explicitly scoped fixed keys are preserved as-is.
```
[the remark this answers is not in the document]
Comment (larrylhollan) [reply packages/adapters/openclaw-gateway/README.md]: Implemented — expanded the README note to call out the `fixed`-strategy behavior too: unscoped fixed keys are auto-prefixed for non-main agents, which can break session continuity on first deploy, so operators may want to drain in-flight sessions first.
